grafana-plugin/src/containers/RemoteFilters/RemoteFilters.tsx

```diff
@@ -316,19 +317,37 @@ class _RemoteFilters extends Component<RemoteFiltersProps, RemoteFiltersState> {
       case 'daterange':
         const dates = values[filter.name] ? values[filter.name].split('/') : undefined;
 
-        const value = {
-          from: dates ? moment(dates[0] + 'Z') : undefined,
-          to: dates ? moment(dates[1] + 'Z') : undefined,
-          raw: {
-            from: dates ? dates[0] : '',
-            to: dates ? dates[1] : '',
-          },
-        };
+        let value = { from: undefined, to: undefined, raw: { from: '', to: '' } };
+        if (dates) {
+          const [fromString, toString] = dates;
+          const isRelative = rangeUtil.isRelativeTimeRange({ from: fromString, to: toString });
+
+          if (isRelative) {
+            const absolute = convertRelativeToAbsoluteDate(values[filter.name]);
+            const [absoluteFrom, absoluteTo] = absolute.split('/');
+            value = {
+              from: moment(absoluteFrom + 'Z'),
+              to: moment(absoluteTo + 'Z'),
+              raw: {
+                from: fromString,
+                to: toString,
+              },
+            };
+          } else {
+            value = {
+              from: moment(fromString + 'Z'),
+              to: moment(toString + 'Z'),
+              raw: {
+                from: fromString,
+                to: toString,
+              },
+            };
+          }
+        }
```

Comment:
Can we extract this part to the helper function like `getValueForDateRangeFilterType`, place it in RemoteFilters.helpers and have it unit tested?

Also seems you could create:
```
const raw = {
                from: fromString,
                to: toString,
              }
```

and reuse it in both conditions to avoid duplication